File: template-default/src/main/java/net/spydroid/template_default/presentation/HomeScreen.kt
/*
 * Copyright (C) 2024 Daniel Gómez(Less)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

package net.spydroid.template_default.presentation

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Text
import androidx.compose.material3.TextField
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.viewmodel.compose.viewModel

@Composable
fun HomeScreen(homeViewModel: HomeViewModel = hiltViewModel()) {

    val valueTask = homeViewModel.newTask.collectAsState().value

    Column(
        Modifier
            .fillMaxSize(1F)
            .padding(10.dp)
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            CreateTask(valueTask) {
                homeViewModel.addNewTextToTask(it)
            }

            Spacer(modifier = Modifier.width(10.dp))
            AddTask {
                homeViewModel.createTask()
            }

        }

    }
}

@Composable
private fun CreateTask(value: String, onChangeValue: (String) -> Unit) {

    OutlinedTextField(
        value = value, onValueChange = {
            onChangeValue(it)
        },
        modifier = Modifier
            .fillMaxWidth(0.7F)
            .height(120.dp),
        shape = RoundedCornerShape(15.dp)
    )
}

@Composable
private fun AddTask(onClick: () -> Unit) {
    Button(
        onClick = { onClick() }, colors = ButtonDefaults.buttonColors(
            containerColor = Color(0xFF380808)
        )
    ) {
        Text(
            text = "Agregar", style = TextStyle(
                color = Color(0xE5FFC5C5),
                fontWeight = FontWeight.SemiBold,
                fontSize = 12.75.sp
            )
        )
    }
}

File: app/src/main/net/spydroid/desktop/presentation/HomeScreen.cpp
/*
 * Copyright (C) 2024 Daniel Gómez(Less)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */
#include "HomeScreen.h"
#include <iostream>
#include <QPainter>
#include "../components/ItemBoard.h"
#include <QDebug>

HomeScreen::HomeScreen(QWidget *parent) : QWidget(parent) {
    
    this->setMinimumSize(600, 500);
    QPalette pal = this->palette();
    pal.setColor(QPalette::Window, QColor("#000000"));
    this->setAutoFillBackground(true);
    this->setPalette(pal);
    
    // Crear el layout para HomeScreen
    layout = new QVBoxLayout(this);

    // Ajustar los márgenes del layout para que los elementos se posicionen
    layout->setContentsMargins(20, 20, 20, 20);  // Márgenes ajustados (izquierda, arriba, derecha, abajo)

    // Crear un widget contenedor para los ItemBoard
    QWidget *itemBoardContainer = new QWidget(this);
    QHBoxLayout *itemBoardLayout = new QHBoxLayout(itemBoardContainer);

    itemBoardLayout->setAlignment(Qt::AlignLeft | Qt::AlignTop); // Alineación en la parte superior izquierda
    itemBoardLayout->setSpacing(10);  // Espaciado entre los widgets

    // Crear el componente ItemBoard y configurarlo
    ItemBoard *dashBoard = new ItemBoard(itemBoardContainer);
    dashBoard->setImage(QPixmap(":/icons/home.png"));
    dashBoard->setText("Panel");
    connect(dashBoard, &ItemBoard::clicked, [this]() {
        emit goToDashBoard();
    });

    ItemBoard *apkStudio = new ItemBoard(itemBoardContainer);
    apkStudio->setImage(QPixmap(":/icons/builder.png"));
    apkStudio->setText("APK Estudio");
    connect(apkStudio, &ItemBoard::clicked, [this]() {
        emit goToBuildApk();
    });


    // Añadir los ItemBoard al layout del contenedor
    itemBoardLayout->addWidget(dashBoard);
    itemBoardLayout->addWidget(apkStudio);

    // Aplicar el layout al contenedor
    itemBoardContainer->setLayout(itemBoardLayout);

    // Añadir el contenedor al layout principal de HomeScreen
    layout->addWidget(itemBoardContainer, 0, Qt::AlignLeft | Qt::AlignTop);

    // Aplicar el layout principal al widget
    setLayout(layout);
}


void HomeScreen::paintEvent(QPaintEvent *event) {
    QPainter painter(this);
    painter.setRenderHint(QPainter::Antialiasing);  // Activar suavizado de bordes

    // Dibujar la imagen de fondo centrada (sin cambios)
    QPixmap background(":/icons/background.png");
    QSize scaledSize = background.size().scaled(800, 800, Qt::KeepAspectRatio);
    QRect targetRect((width() - scaledSize.width()) / 2, (height() - scaledSize.height()) / 2, scaledSize.width(), scaledSize.height());
    QPixmap scaledPixmap = background.scaled(scaledSize, Qt::KeepAspectRatio, Qt::SmoothTransformation);
    painter.drawPixmap(targetRect, scaledPixmap);

    // Establecer el color y grosor del borde rojo
    QPen pen(QColor("#FF0000"));  // Color rojo para el borde
    pen.setWidth(4);  // Grosor del borde
    painter.setPen(pen);

    // Establecer un brush transparente para que solo se vea el borde
    QBrush brush(Qt::NoBrush);
    painter.setBrush(brush);

    // Dibujar un rectángulo redondeado con padding de 20px (para que no toque los bordes)
    int padding = 15;
    painter.drawRoundedRect(padding, padding, width() - 2 * padding, height() - 2 * padding, 20, 20);  // Bordes redondeados de 20px

    // Llamar al método base para asegurar que el evento de pintura continúe normalmente
    QWidget::paintEvent(event);
}
